lib/millstone.js

```diff
@@ -443,8 +445,32 @@ function unzip(file, callback) {
                 path.basename(file, path.extname(file)) +
                 path.extname(name).toLowerCase()
             );
-            zf.copyFile(name, dest, function(err) {
-                if (err) return callback(err);
+            try {
+                // We implement here the writting based on adm-zip writeFileTo
+                // because extractEntryTo doesn't allow to change the destination name (just the folder)
+                const content = zipFile.getData();
+                const folder = path.dirname(dest);
+                if (!fs.existsSync(folder)) {
```

Comment:
we cannot use `*Sync` methods in production (only allowed on the module's top-level  as they only are executed once, at startup time) as they are CPU bound and block the main event loop.

My suggestions are:

- a) requires a breaking change as `promisify` and `async/await` are not available in Node.js 6
```js
const { promisify } = require('util');
const fs = require('fs');
const existsPath = promisify(fs.exists);
const mkdir = promisisfy(fs.mkdir);
const open = promisisfy(fs.open);
[....] 
const folder = path.dirname(dest);
const exists = await existsPath(folder)
if (!exists(folder)) {
    await mkdir(folder)
}
[...]
fd = await open(dest, 'w', 438);
[...]
```
b) Use the callback's version of fs module (does not require a major version)
c) Use https://nodejs.org/dist/latest-v10.x/docs/api/fs.html#fs_fs_promises_api; requires a to drop support for Node.js < 10

Reply:
I'll go with b for now, which should be "easy" based on https://github.com/cthackers/adm-zip/blob/f66dc607e0c6f31affa0ff75461537f77296b4ef/util/utils.js#L136.

Reply:
Done. There are other sync calls within the module, but I think we can leave that for the next refactor iteration.